indico/modules/events/contributions/controllers/display.py

```diff
@@ -0,0 +1,165 @@
+        'locators': {
+            lambda self: self.author
+        }
+    }
+
+    def _checkParams(self, params):
+        RHContributionDisplayBase._checkParams(self, params)
+        self.author = (ContributionPersonLink
+                       .find(contribution=self.contrib, person_id=request.view_args['person_id'])
+                       .filter(~ContributionPersonLink.author_type.in_([AuthorType.none]))
+                       .one())
+
+    def _process(self):
+        author_contribs = (Contribution.query.with_parent(self.event_new)
+                           .join(ContributionPersonLink)
+                           .options(noload('*'))
+                           .options(joinedload('event_new'))
+                           .options(load_only('id', 'title'))
+                           .filter(ContributionPersonLink.person_id == self.author.person_id, ~Contribution.is_deleted,
+                                   ~ContributionPersonLink.author_type.in_([AuthorType.none]))
+                           .all())
+        return WPContributions.render_template('display/contribution_person.html', self._conf,
+                                               author=self.author.person, contribs=author_contribs)
+
+
+class RHContributionExportToPDF(RHContributionDisplayBase):
+    def _process(self):
+        pdf = ContribToPDF(self.contrib)
+        return send_file('contribution.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHContributionsExportToPDF(RHContributionList):
+    def _process(self):
+        contribs = self.reporter.get_contrib_report_kwargs()['contribs']
+        pdf = ContribsToPDF(self._conf, contribs)
+        return send_file('contributions.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHContributionExportToICAL(RHContributionDisplayBase):
+    """Export contribution to ICS"""
+
+    def _process(self):
+        data = {'results': serialize_contribution(self.contrib)}
+        serializer = Serializer.create('ics')
+        return send_file('contribution.ics', BytesIO(serializer(data)), 'text/calendar')
+
+
+class RHContributionReport(RHContributionList):
+    """Display dialog with filters"""
+
+    def _process(self):
+        return RH._process(self)
+
+    def _process_GET(self):
+        return WPContributions.render_template('contrib_report_filter.html', self._conf, event=self.event_new,
+                                               filters=self.reporter.report_config['filters'],
+                                               filterable_items=self.reporter.filterable_items)
+
+    def _process_POST(self):
+        self.reporter.store_filters()
+        return jsonify_data(**self.reporter.render_contribution_list())
+
+
+class RHContributionListStaticURL(RHContributionList):
+    """Generate static URL for the current set of filters"""
+
+    def _process(self):
+        return jsonify(url=self.reporter.generate_static_url())
+
+
+class RHSubcontributionDisplay(RHConferenceBaseDisplay):
```

Comment:
url normalization (`normalize_url_spec`) is missing! so right now you could access any subcontrib through an event you manage.

make sure that all rhs which get an object just by its ID normalize the URL to prevent this